src/screens/Main/Main.screen.tsx

```diff
@@ -0,0 +1,42 @@
+import React, {FC, useEffect} from 'react';
+import {Text, ScrollView} from 'react-native';
+import {useSelector, useDispatch} from 'react-redux';
+import {publicChannelsSelectors} from '../../store/publicChannels/publicChannels.selectors';
+import {publicChannelsActions} from '../../store/publicChannels/publicChannels.slice';
+import {IChannelInfo} from '../../store/publicChannels/publicChannels.types';
+
+export const MainScreen: FC = () => {
+  const channels = useSelector(publicChannelsSelectors.publicChannels);
+  const messages = useSelector(publicChannelsSelectors.currentChannelMessages);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(publicChannelsActions.getPublicChannels());
+  }, [dispatch]);
+
+  useEffect(() => {
```

Comment:
I'd have entire logic of this effect in saga that loaded channels

Reply:
That's the same thing that I mentioned above. I think we should discuss it more and think about possible solutions with WaggleMobileManager